doc/source/administrator/authentication.md

```diff
@@ -0,0 +1,427 @@
+        state: state
+    JupyterHub:
+      authenticator_class: oauthenticator.generic.GenericOAuthenticator
+```
+
+#### Auth0
+
+Auth0 is a popular commercial provider of identity management. The
+JupyterHub helm chart does not include support for Auth0 by default. To
+use Auth0, `extraEnv` and `extraConfig` must be configured as follows:
+
+Note that without the scope defined, authenticating to JupyterHub after
+already being logged in to Auth0 will fail.
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: client-id-from-auth0-here
+      client_secret: client-secret-from-auth0-here
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+      scope:
+        - openid
+        - email
+    Auth0OAuthenticator:
+      auth0_subdomain: prod-8ua-1yy9
+    Authenticator:
+      admin_users:
+        - devops@example.com
+      auto_login: true
+    JupyterHub:
+      authenticator_class: oauthenticator.auth0.Auth0OAuthenticator
+```
+
+#### Full Example of Google OAuth2
+
+If your institution is a [G Suite customer](https://gsuite.google.com)
+that integrates with Google services such as Gmail, Calendar, and Drive,
+you can authenticate users to your JupyterHub using Google for
+authentication.
+
+1.  Log in to the [Google API
+    Console](https://console.developers.google.com).
+2.  Select a project > Create a project\... and set 'Project name'.
+    This is a short term that is only displayed in the console. If you
+    have already created a project you may skip this step.
+3.  Type "Credentials" in the search field at the top and click to
+    access the Credentials API.
+4.  Click "Create credentials", then "OAuth client ID". Choose
+    "Application type" > "Web application".
+5.  Enter a name for your JupyterHub instance. You can give it a
+    descriptive name or set it to be the hub's hostname.
+6.  Set "Authorized JavaScript origins" to be your hub's URL.
+7.  Set "Authorized redirect URIs" to be your hub's URL followed by
+    `/hub/oauth_callback`. For example,
+    `https://your-jupyterhub-domain/hub/oauth_callback`.
+8.  When you click "Create", the console will generate and display a
+    Client ID and Client Secret. Save these values.
+9.  Type "consent screen" in the search field at the top and click to
+    access the OAuth consent screen. Here you will customize what your
+    users see when they login to your JupyterHub instance for the first
+    time. Click Save when you are done.
+10. In your helm chart, create a stanza that contains these OAuth
+    fields:
+
+```yaml
+hub:
+  config:
+    OAuthenticator:
+      client_id: your-client-id.apps.googleusercontent.com
+      client_secret: your-client-secret
+      oauth_callback_url: https://your-jupyterhub-domain/hub/oauth_callback
+    GoogleOAuthenticator:
+      hosted_domain:
+        - your-university.edu
+      login_service: Your university
+    JupyterHub:
+      authenticator_class: oauthenticator.GoogleOAuthenticator
```

Comment:
In other places you've used `oauthenticator.MODULE.XXXAuthenticator` instead of `oauthenticator.XXXAuthenticator`. Either works, but I think it'd be good to be consistent. Alternatively can an entrypoint be specified?

Reply:
@manics oh, I have just copied the jupyterhub_config.py files mapping. I assume this relates to the __init__.py file of the oauthenticator modules folder? I suggest we keep it short but not using `google` or similar as it hides the fact its related to the oauthenticator python package which I want us to highlight.

Reply:
https://github.com/jupyterhub/oauthenticator/blob/6f239bebecbb3fb0242de7f753ae1c93ed101340/oauthenticator/__init__.py#L1-L7

It seems that only a few are added there for backward compatebility. So, we can only make all longer. Let's make all longer.

Reply:
Done.